compiler/rustc_middle/src/mir/interpret/pointer.rs

```diff
@@ -120,9 +120,14 @@ pub trait Provenance: Copy + fmt::Debug {
     where
         Self: Sized;
 
-    /// Provenance must always be able to identify the allocation this ptr points to.
+    /// Whether this provenance specfies a non-contiguous (permissive) provenance. Can only return true if `OFFSET_IS_ADDR` is true.
+    fn is_permissive(self) -> bool;
```

Comment:
Is this the same as what we often called a "wildcard" provenance? I think that would be a better term to use here.